state/application_test.go

```diff
@@ -254,6 +254,8 @@ func (s *ApplicationSuite) TestSetCharmCharmSettings(c *gc.C) {
 }
 
 func (s *ApplicationSuite) TestGenerationCharmCharmSettings(c *gc.C) {
+	c.Skip("To be rewritten when branch-based configuration reads are implemented.")
```

Comment:
I will reinstate this as a set/get test when reading branch-based charm configuration is implemented in a following patch.